### PR TITLE
1192: update case trial sort tags when case is updated

### PR DESCRIPTION
**Pull request by rachaelparris**
This is dependent on #2034 - that one should be merged first.

### CODE REVIEW

File: shared/src/persistence/dynamo/cases/updateCaseTrialSortMappingRecords.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+exports.updateCaseTrialSortMappingRecords = async ({
+  caseId,
+  caseSortTags,
+  applicationContext,
+}) => {
+  const { hybrid, nonHybrid } = caseSortTags;
+
+  const oldSortRecords = await client.query({
+    ExpressionAttributeNames: {
```
Comment (codyseibert) [inline shared/src/persistence/dynamo/cases/updateCaseTrialSortMappingRecords.js:21]: I think you'd need to provide an `IndexName: 'gsi1'` for this to work

File: shared/src/persistence/dynamo/cases/updateCaseTrialSortMappingRecords.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    ExpressionAttributeValues: {
+      ':gsi1pk': `eligible-for-trial-case-catalog-{caseId}`,
+      ':pk': 'eligible-for-trial-case-catalog',
+    },
+    KeyConditionExpression: '#gsi1pk = :gsi1pkpk AND #pk = :pk',
+    applicationContext,
+  });
+
+  if (
+    oldSortRecords[0] &&
```
Comment (codyseibert) [inline shared/src/persistence/dynamo/cases/updateCaseTrialSortMappingRecords.js]: oldSortRecords might be an empty array I think, so you might want to add `oldSortRecords.length &&` unless you don't care this endpoint might fail

File: shared/src/persistence/dynamo/cases/updateCaseTrialSortMappingRecords.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    (oldSortRecords[0].sk !== hybrid && oldSortRecords[0].sk !== nonHybrid)
+  ) {
+    deleteCaseTrialSortMappingRecords({
```
Comment (codyseibert) [inline shared/src/persistence/dynamo/cases/updateCaseTrialSortMappingRecords.js]: does this need an `await`? or is this good.

Comment (rachaelparris) [reply shared/src/persistence/dynamo/cases/updateCaseTrialSortMappingRecords.js]: Sure does. Thanks!

**Review by wbyoko — APPROVED** [review]

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**